src/main/java/com/minecolonies/coremod/colony/managers/CitizenManager.java

```diff
@@ -110,28 +110,23 @@ public void writeToNBT(@NotNull final NBTTagCompound compound)
 
     @Override
     public void sendPackets(
-      @NotNull final Set<EntityPlayerMP> oldSubscribers,
-      final boolean hasNewSubscribers,
-      @NotNull final Set<EntityPlayerMP> subscribers)
+      @NotNull final Set<EntityPlayerMP> closeSubscribers,
+      @NotNull final Set<EntityPlayerMP> newSubscribers)
     {
-        if (isCitizensDirty || hasNewSubscribers)
+        if (isCitizensDirty || !newSubscribers.isEmpty())
         {
+            final Set<EntityPlayerMP> players = isCitizensDirty ? closeSubscribers : newSubscribers;
             for (@NotNull final ICitizenData citizen : citizens.values())
             {
                 if (citizen.getCitizenEntity().isPresent())
                 {
-                    if (citizen.isDirty() || hasNewSubscribers)
+                    if (citizen.isDirty() || !newSubscribers.isEmpty())
                     {
-                        subscribers.stream()
-                          .filter(player -> citizen.isDirty() || !oldSubscribers.contains(player))
-                          .forEach(player -> MineColonies.getNetwork().sendTo(new ColonyViewCitizenViewMessage(colony, citizen), player));
+                        players.forEach(player -> MineColonies.getNetwork().sendTo(new ColonyViewCitizenViewMessage(colony, citizen), player));
```

Comment:
I think there is a network message to send it to a list of players, isn't there?